example/lib/develop/peripheral_list/peripheral_list_bloc.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:async';
+import 'package:blemulator_example/develop/model/ble_peripheral.dart';
+import 'package:blemulator_example/develop/repository/peripheral_list_repository.dart';
+import 'package:bloc/bloc.dart';
+import './bloc.dart';
+
+class PeripheralListBloc
+    extends Bloc<PeripheralListEvent, PeripheralListState> {
+  List<BlePeripheral> _peripherals = <BlePeripheral>[];
+  bool _scanningEnabled = false;
+  PeripheralListRepository _peripheralListRepository;
+
+  PeripheralListBloc(this._peripheralListRepository);
+
+  @override
+  PeripheralListState get initialState =>
+      PeripheralListState(_peripherals, _scanningEnabled);
+
+  @override
+  Stream<PeripheralListState> mapEventToState(
+    PeripheralListEvent event,
+  ) async* {
+    if (event is StartPeripheralScan) {
+      yield* _mapStartPeripheralScanToState(event);
+    } else if (event is StopPeripheralScan) {
+      yield* _mapStopPeripheralScanToState(event);
+    } else if (event is NewPeripheralScan) {
+      yield* _mapNewPeripheralScanToState(event);
```

Comment:
imho it is unintuitive that in the same place we are handling internal and external events

Reply:
Hmm I can see your point, but I wouldn't say that NewPeripheralscan is an internal event - it's being added to the sink by the repository in the ScanEventOutputFunction callback of it's startPeripheralScan method.

Reply:
Created a separate issue for changing the implementation of NewPeripheralScan event.